server/core/email.js

```diff
@@ -82,6 +87,62 @@ async function sendVerificationEmail(email, username) {
   return await sendMail(email, subject, html);
 }
 
+async function sendResetPasswordEmail(email, user) {
+  const { username, resetPasswordToken } = user;
+  const token = jwt.sign(
+    { username, resetPasswordToken }, // payload
+    JWT_SECRET, // secret
+    { expiresIn: RESET_PASSWORD_EXPIRATION } // options
+  );
+
+  const subject = 'Reset Password Request';
+  const url = `https://www.watsmymajor.com/reset-password?token=${token}`;
+  const html = `
+  ${emailStyles}
+    <h2>Hi ${username},</h2>
+    <h3>
+      We received a request to reset your WatsMyMajor account password.
+    </h3>
+    <a href="${url}" id="button">Reset Password</a>
+    <p>
+      If you’re having trouble clicking the button, copy and paste the URL below into your web browser.
+    </p>
+    <a href="${url}">${url}</a>
+    <p>
+      If you didn't request a password reset, please ignore this email or <a href="https://github.com/theRoughCode/watsmymajor/issues">let us know</a>.
+    </p>
+  `;
+
+  return await sendMail(email, subject, html);
+}
+
+async function sendResetPasswordSuccess(email, username) {
+  const subject = 'Reset Password Successful';
+  const html = `
+    <h2>Hi ${username},</h2>
+    <h3>
+      Your password was successfully updated!
+    </h3>
+    <p>
+      If you didn't request a password reset, <a href="https://github.com/theRoughCode/watsmymajor/issues">contact us</a>.
+    </p>
+  `;
+
+  return await sendMail(email, subject, html);
+}
+
+// Verify email token to verify user's email
+// Returns { err, username, resetPasswordToken }
+async function verifyResetPasswordToken(token) {
+  try {
+    const { username, resetPasswordToken } = jwt.verify(token, JWT_SECRET);
+    return { err: null, username, resetPasswordToken };
+  } catch (err) {
+    console.error(err);
+    return { err, username: null };
```

Comment:
add in resetPasswordToken here to ensure same object fields being returned (minor optimization).